cmd/snap-update-ns/utils.go

```diff
@@ -89,31 +89,32 @@ func (sec *Secure) CheckTrespassing(fd int, segments []string, segNum int) error
 // The file descriptor is opened using the O_PATH, O_NOFOLLOW,
 // and O_CLOEXEC flags.
 func (sec *Secure) OpenPath(path string) (int, error) {
-	if !filepath.IsAbs(path) {
-		return -1, fmt.Errorf("path %v is not absolute", path)
-	}
-	segments, err := splitIntoSegments(path)
+	iter, err := NewPathIterator(path)
 	if err != nil {
-		return -1, err
+		// TODO: Reword the error and adjust the tests.
```

Comment:
This is fine. I guess you'll do it in a followup PR?

Reply:
Yes, I already have it pushed :)